library/README.md

```diff
@@ -119,35 +125,90 @@ NDK_HOME=$HOME/Documents/GitHub/engine/src/third_party/android_tools/ndk cargo n
 - Updates must be applied in order.
 - Updates are applied in a single transaction.
 
-### Update State Machine
+### State Machines
+
+#### Boot State Machine
+
+This state machine tracks the process of the engine starting up, with or without
+a patch. These state transitions happen whether or not a new patch is available,
+but in the context of the updater, we only care about the case where we are
+booting from a patch.
+
+The patch boot state is internal to the PatchManager and stored on disk in
+`patches_state.json`. It contains three fields, all of which are Optional
+PatchMetadata structs:
+
+- last_boot_patch: The last patch that was successfully booted.
+- last_attempted_patch: The last patch that we attempted to boot.
+- next_boot_patch: The next patch that we will attempt to boot.
+
+This state machine has the following states. It can only move forward through
+them.
+
+1. Ready - The engine is initialized but has not started booting yet.
+2. Booting - The engine has started booting.
+3. Success - The engine successfully booted.
+4. Failure - The engine failed to boot.
```

Comment:
Is this actually recorded in an enum?  (if not that might be a next step, or worth commenting).

Reply:
Added an assumptions list at the bottom and specifically called out that PatchManager state is implicit, but should be more explicit.